### PR TITLE
[AMDAIETemporaryAllocBufferization] Bufferize temporary allocs that aren't defined in core

### PR DESCRIPTION
**Pull request by newling**
Update the pass to handle `memref.alloc` in any scope, not just in `amdaie.core`

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/temporary_alloc_bufferization.mlir [hunk 1]
```diff
@@ -86,3 +76,34 @@ func.func @temp_buffer() {
...
+//   CHECK-DAG:   %[[C0:.*]] = arith.constant 0 : index
+//   CHECK-DAG:   %[[C1:.*]] = arith.constant 1 : index
+//   CHECK-DAG:   %[[TILE_0:.*]] = amdaie.tile(%[[C0]], %[[C0]])
+//   CHECK-DAG:   %[[BUFFER_0:.*]] = amdaie.buffer(%[[TILE_0]]) : memref<4xi32>
+//   CHECK-DAG:   %[[TILE_1:.*]] = amdaie.tile(%[[C0]], %[[C1]])
+//   CHECK-DAG:   %[[BUFFER_1:.*]] = amdaie.buffer(%[[TILE_1]]) : memref<4xi32>
+//       CHECK:   amdaie.core(%[[TILE_0]]
+//       CHECK:     %[[CAST:.*]] = memref.reinterpret_cast %[[BUFFER_0]]
+//       CHECK:   amdaie.core(%[[TILE_1]]
+//       CHECK:     %[[CAST:.*]] = memref.reinterpret_cast %[[BUFFER_1]]
```
Comment (jtuyls) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/temporary_alloc_bufferization.mlir:92]: I would prefer for style consistency that we don't indent the `CHECK` and place them one space after `//` as we do in the rest of the codebase (except for this file apparently :P).

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETemporaryAllocBufferization.cpp [hunk 2]
```diff
@@ -13,47 +13,42 @@ namespace mlir::iree_compiler::AMDAIE {
...
-    SmallVector<Operation *> &toBeErased) {
-  // Step 1. Get all buffers within a CoreOp.
-  SmallVector<memref::AllocOp> allocOps;
-  coreOp.walk([&](Operation *op) {
-    if (auto allocOp = dyn_cast<memref::AllocOp>(op)) {
-      allocOps.push_back(allocOp);
-      toBeErased.push_back(allocOp);
-    } else if (auto deallocOp = dyn_cast<memref::DeallocOp>(op)) {
-      toBeErased.push_back(deallocOp);
+LogicalResult bufferizeTemporaryMemrefs(Operation *parentOp) {
```
Comment (jtuyls) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETemporaryAllocBufferization.cpp:16]: While this is an improvement, I think we should go from `access` directly to `amdaie.buffer` and avoid the alloc intermediary.

**Review by jtuyls — APPROVED** [review]
LGTM

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETemporaryAllocBufferization.cpp (hunk 2 above)
Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETemporaryAllocBufferization.cpp:16]: Yes ok I think that makes sense, I'll pick that up next.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/temporary_alloc_bufferization.mlir (hunk 1 above)
Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/temporary_alloc_bufferization.mlir:92]: Noted. :P

**Pull request merged**